src/compiler/checker.ts

```diff
@@ -1322,8 +1322,14 @@ namespace ts {
                             }
                         }
 
+                        // ES6 exports are also visible locally (except for 'default'), but commonjs exports are not (except typedefs)
                         if (name !== InternalSymbolName.Default && (result = lookup(moduleExports, name, meaning & SymbolFlags.ModuleMember))) {
-                            break loop;
+                            if ((location as SourceFile).commonJsModuleIndicator && !result.declarations.some(isJSDocTypeAlias)) {

```

Comment:
May be better to do this check before calling `lookup`.

Reply:
Can you also check that the location is a soruceFile

Reply:
sure, although it would be really weird for a ModuleDeclaration to have a property named commonJsModuleIndicator

Reply:
@andy-ms I still want to call lookup, even for commonjs modules, because it might find a jsdoc type alias. I can't unconditionally skip commonjs modules.

Reply:
(believe me, I tried :)